cli/fmt_errors.rs

```diff
@@ -220,25 +228,59 @@ fn format_js_error(js_error: &JsError, is_child: bool) -> String {
 /// Wrapper around deno_core::JsError which provides colorful
 /// string representation.
 #[derive(Debug)]
-pub struct PrettyJsError(JsError);
+pub struct PrettyJsError {
+  js_error: JsError,
+  color_message: bool,
+}
 
 impl PrettyJsError {
   pub fn create(js_error: JsError) -> AnyError {
-    let pretty_js_error = Self(js_error);
+    let pretty_js_error = Self {
+      js_error,
+      color_message: false,
+    };
+    pretty_js_error.into()
+  }
+
+  pub fn create_for_testing(mut js_error: JsError) -> AnyError {
+    // filter out stack frames that are internal Deno code (starting with
+    // "[deno:" or "deno:"
+    let frames = std::mem::take(&mut js_error.frames);
+    let frames = frames
+      .into_iter()
+      .rev()
+      .skip_while(|f| {
+        if let Some(file_name) = &f.file_name {
+          file_name.starts_with("[deno:") || file_name.starts_with("deno:")
+        } else {
+          false
+        }
+      })
+      .collect();
+    js_error.frames = frames;
+
+    let pretty_js_error = Self {
+      js_error,
+      color_message: true,
+    };
```

Comment:
@nayeemrmn does this seem fine?

Reply:
IMO this change shouldn't involve any change to `fmt_errors.rs`, we should have a function in `cli/tools/test.rs`:
```rust
fn format_test_error(js_error: &JsError) -> String {
  let mut js_error = js_error.clone();
  let e = colors::red(&js_error.exception_message).to_string();
  js_error.exception_message = e;
  let frames = std::mem::take(&mut js_error.frames);
  let frames = frames
    .into_iter()
    .rev()
    .skip_while(|f| {
      if let Some(file_name) = &f.file_name {
        file_name.starts_with("[deno:") || file_name.starts_with("deno:")
      } else {
        false
      }
    })
    .collect();
  js_error.frames = frames;
  PrettyJsError::create(js_error)
    .to_string()
    .trim_start_matches("Uncaught ")
    .to_string()
}
```
```rust
for (description, js_error) in &summary.failures {
  // ...
  println!("{}", format_test_error(js_error));
  println!();
}
```

Reply:
Makes sense, I'll do that in separate PR, so this change can be landed independently.

Reply:
https://github.com/denoland/deno/pull/14302